src/utils/validationFunctions.ts

```diff
@@ -0,0 +1,19 @@
+import { findUserByEmail, findUserByUsername } from "../models/user";
+
+export const isUsernameTaken = async (value: string) => {
+  const foundUser = await findUserByUsername(value);
+  if (!foundUser) {
+    return Promise.resolve("Success");
```

Comment:
This function is `async`, meaning it'll always return a promise.

`return 'Success'` would have exactly same effect.

But aside from that, returning strings is not a good way of communicating success/failure.

I think this function could just be this:

```
export const isUsernameTaken = async (value: string) => {
  const foundUser = await findUserByUsername(value);
  if (foundUser) {
    throw new Error('Username already in use');
  }
};
```